cpp/src/rolling/rolling_detail.hpp

```diff
@@ -39,25 +39,39 @@ namespace detail
 
     constexpr bool timestamp_mean =
       is_mean &&
-      std::is_same<AggOp, DeviceSum>::value &&
       cudf::is_timestamp<ColumnType>();
 
+    constexpr bool timestamp_sum =
+        std::is_same<AggOp, DeviceSum>::value &&
+        cudf::is_timestamp<ColumnType>();
+
+
     constexpr bool is_valid_rolling_agg = !std::is_same<ColumnType, cudf::string_view>::value and
```

Comment:
Getting complicated. Maybe a doc block enumerating exactly what is supported. 

The do a separate `constexpr bool` variable for each type category, e.g.

`is_valid_string_agg`, `is_valid_timestamp_agg`, `is_valid_numeric_agg`. Then you can just or them together.